src/dune/action_unexpanded.ml

```diff
@@ -499,11 +499,19 @@ module Infer = struct
             acc
       end)
 
-  let partial ~all_targets t =
-    if all_targets then
-      Partial_with_all_targets.infer t
-    else
-      Partial.infer t
+  let partial (targets : Targets.Or_forbidden.t) t =
+    match targets with
+    | Targets Infer -> Partial_with_all_targets.infer t
+    | Forbidden _ ->
+      (* XXX why don't we make sure that no targets were inferred? *)
```

Comment:
I wondered about it when I initially wrote this code, but since our target detection is not super robust and can both detect false targets and miss some, I decided to not try and be strict here.

Reply:
I added this to the comment.